pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -220,61 +218,53 @@ func (pl *PodTopologySpread) calPreFilterState(pod *v1.Pod) (*preFilterState, er
 	s := preFilterState{
 		Constraints:          constraints,
 		TpKeyToCriticalPaths: make(map[string]*criticalPaths, len(constraints)),
-		TpPairToMatchNum:     make(map[topologyPair]*int32, sizeHeuristic(len(allNodes), constraints)),
+		TpPairToMatchNum:     make(map[topologyPair]int32, sizeHeuristic(len(allNodes), constraints)),
 	}
 
-	// Nodes that pass nodeAffinity check and carry all required topology keys will be
-	// stored in `filteredNodes`, and be looped later to calculate preFilterState.
-	var filteredNodes []*framework.NodeInfo
 	requiredSchedulingTerm := nodeaffinity.GetRequiredNodeAffinity(pod)
-	for _, n := range allNodes {
-		node := n.Node()
+	tpCountsByNode := make([]map[topologyPair]int32, len(allNodes))
+	processNode := func(i int) {
+		nodeInfo := allNodes[i]
+		node := nodeInfo.Node()
 		if node == nil {
 			klog.ErrorS(nil, "Node not found")
-			continue
+			return
 		}
 		// In accordance to design, if NodeAffinity or NodeSelector is defined,
 		// spreading is applied to nodes that pass those filters.
 		// Ignore parsing errors for backwards compatibility.
 		match, _ := requiredSchedulingTerm.Match(node)
 		if !match {
-			continue
+			return
 		}
 		// Ensure current node's labels contains all topologyKeys in 'Constraints'.
 		if !nodeLabelsMatchSpreadConstraints(node.Labels, constraints) {
-			continue
+			return
 		}
+
+		tpCounts := make(map[topologyPair]int32, len(constraints))
 		for _, c := range constraints {
 			pair := topologyPair{key: c.TopologyKey, value: node.Labels[c.TopologyKey]}
-			s.TpPairToMatchNum[pair] = new(int32)
+			count := countPodsMatchSelector(nodeInfo.Pods, c.Selector, pod.Namespace)
+			tpCounts[pair] = int32(count)
 		}
-
-		filteredNodes = append(filteredNodes, n)
+		tpCountsByNode[i] = tpCounts
 	}
+	pl.parallelizer.Until(context.Background(), len(allNodes), processNode)
```

Comment:
(this code pre-exists)

Can you help to pass in `ctx` down here? 

````
calPreFilterState(ctx context.Context, pod *v1.Pod)
```

Reply:
absolutely, updated.

Reply:
LGTM overall. Could you squash the commits?